src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs

```diff
@@ -573,60 +533,69 @@ await WriteAsync(serverSslStream, new byte[] { 1 }, 0, 1)
         [Fact]
         public void SslStream_StreamToStream_Flush_Propagated()
         {
-            VirtualNetwork network = new VirtualNetwork();
-
-            using (var stream = new VirtualNetworkStream(network, isServer: false))
-            using (var sslStream = new SslStream(stream, false, AllowAnyServerCertificate))
+            var ms = new MemoryStream();
+            var tracking = new CallTrackingStream(ms);
+            using (var sslStream = new SslStream(tracking, false, AllowAnyServerCertificate))
             {
-                Assert.False(stream.HasBeenSyncFlushed);
+                Assert.Equal(0, tracking.TimesCalled(nameof(Stream.Flush)));
                 sslStream.Flush();
-                Assert.True(stream.HasBeenSyncFlushed);
+                Assert.NotEqual(0, tracking.TimesCalled(nameof(Stream.Flush)));
             }
         }
 
         [Fact]
-        public void SslStream_StreamToStream_FlushAsync_Propagated()
+        public async Task SslStream_StreamToStream_FlushAsync_Propagated()
         {
-            VirtualNetwork network = new VirtualNetwork();
-
-            using (var stream = new VirtualNetworkStream(network, isServer: false))
+            (Stream stream1, Stream stream2) = ConnectedStreams.CreateBidirectional();
+            var tcs = new TaskCompletionSource();
+            using (var stream = new DelegateDelegatingStream(stream1) { FlushAsyncFunc = async cancellationToken => { await tcs.Task.WithCancellation(cancellationToken); await stream1.FlushAsync(cancellationToken); } })
             using (var sslStream = new SslStream(stream, false, AllowAnyServerCertificate))
+            using (stream2)
             {
-                stream.DelayFlush = true;
                 Task task = sslStream.FlushAsync();
 
                 Assert.False(task.IsCompleted);
-                stream.CompleteAsyncFlush();
-                Assert.True(task.IsCompleted);
+                tcs.SetResult();
+
+                await task;
+            }
+
+            var ms = new MemoryStream();
```

Comment:
This code looks like the same code in the sync Flush test above. Did you mean to modify this test to work the same way as the sync one?